server/test/instant/util/cache_test.clj

```diff
@@ -1,129 +1,115 @@
-                       deref
-                       (cache/lu-cache-factory :threshold 100))
-        latch (java.util.concurrent.CountDownLatch. thread-count)
+  (let [thread-count        100
+        cache               (cache/make
+                             {:ttl 120000
+                              :max-size 100})
+        latch               (java.util.concurrent.CountDownLatch. thread-count)
         invocations-counter (atom 0)
-        values (atom [])]
+        values              (atom [])]
     (dotimes [_ thread-count]
       (.start (Thread. (fn []
                          (swap! values conj
-                                (ucache/lookup-or-miss-batch
-                                 cache-atom
+                                (cache/get-all-sync
+                                 cache
                                  ["my-key"]
-                                 (fn [_]
+                                 (fn [keys]
                                    (swap! invocations-counter inc)
                                    (Thread/sleep 1000)
-                                   {"my-key" "some value"})))
+                                   (into {} (for [k keys] [k "some value"])))))
                          (.countDown latch)))))
-
     (.await latch)
     (is (= 1 (deref invocations-counter)))
     (doseq [v @values]
       (is (= {"my-key" "some value"} v)))))
 
 (deftest lookup-or-miss-rejects-errors
-  (let [cache-atom (cache/lru-cache-factory {} :threshold 2)]
-    (testing "demonstrate the problem"
-      ;; Put an error in the cache with the default lookup-or-miss
-      (is (thrown? Exception
-                   (cache/lookup-or-miss cache-atom :wrapped (fn [_]
-                                                               (throw (Exception. "oops"))))))
-
-      ;; Now we'll never get that error out of the cache
-      (is (thrown? Exception
-                   (cache/lookup-or-miss cache-atom :wrapped (fn [_] :ok)))))
-
-
-    (testing "our version evicts errors"
-      (is (thrown? Exception
-                   (ucache/lookup-or-miss cache-atom :instant (fn [_]
-                                                                (throw (Exception. "oops"))))))
+  (let [cache (cache/make {:max-size 2})]
+    (is (thrown? Exception
+                 (cache/get cache :instant (fn [_]
+                                             (throw (Exception. "oops"))))))
 
-      (is (= :instant (ucache/lookup-or-miss cache-atom :instant (fn [x] x)))))))
+    (is (= :instant (cache/get cache :instant (fn [x] x))))))
 
 (deftest lookup-or-miss-batch-rejects-errors
-  (let [cache-atom (cache/lru-cache-factory {} :threshold 2)]
+  (let [cache (cache/make {:max-size 2})]
     (is (thrown? Exception
-                 (ucache/lookup-or-miss-batch cache-atom [:instant] (fn [_]
-                                                                      (throw (Exception. "oops"))))))
+                 (cache/get-all cache [:instant] (fn [_]
```

Comment:
I think this test will fail if `get-all-sync` is called. There's nothing to evict the delay with the error from the cache.